### DISCUSSION TITLE
How to transfer Parameters between pages and modals

### DISCUSSION THREAD

**Discussion by Romska** [Budibase/budibase#3380] [Help] [answered]
Good day! 

First of all some offtop. I have met low-coding platforms recently. I started acquaintance with Appsmith. But it appeared that its module "list" is not ready to work with. Your module "repeater" is far more convenient. In general your platform seemed better than Apssmith as for me, but there is one problem - documentation. Do you have any plans to make it better? What other resourses can I visit to study working with your app (maybe your saw articles at medium, I couldnt find interesting one)?

Can you explain how to transfer parameters between pages and modals? For example, I want to add a button that opens a page, where data from an input is printed, How can I do this? What and where can I read to understand the mechanism? I couldnt find answers in your documentation. May be I was seek. Oh, what is important - I want to do it without usage of url params. 

P.S. I tried to download template examples, but they are not available at your site. The other question is - can your share some examples of working apps, that already have these function realised?

**Comment by samg666** [comment]
use button navigate to something like `/page1/{{param1}}/{{param2}}`, on the page1 you should set the path `/page1/:param1/:param2`, then you can access the params by `url.param1`, `url.param2`.

Btw, I am also transfer from appsmith to budibase, budibase is much better for me on speed, stability, although not that powerful than appsmith yet, but still good enough for me. I will suggest you import the budibase demo apps, then you can learn many from it.

**Comment by shogunpurple** [comment, marked as answer]
Hi @Romska - thank you for the kind words. We are glad you are enjoying using budibase so far.

With regards to documentation, it is something we are constantly working on to improve, and we will have many more tutorials, guides and documentation for more advanced use cases in future. In addition to the documentation, the best way to learn budibase is by looking at some of our existing applications and templates that you can use inside your own builder, available in-builder when you create a new app or from our website here: https://budibase.com/templates

You can keep up with the latest budibase features and releases here on our youtube channel: https://www.youtube.com/c/Budibase

Another good way to find information is in this discussions forum - search through some of the discussions and some other members of our community may have had the same issue or problem that you have.

Now for your use case - @ggs331 suggestion is a good way to do this, but you mentioned you do not want to use URL parameters. I've used our "global state" feature to do this - taking the value from the input, putting it into a key called `myinput`, and displaying it within another screen shown in a modal.

<img width="1638" alt="Screenshot 2021-11-15 at 10 15 49" src="https://user-images.githubusercontent.com/11256663/141754690-e9ed7948-99f2-4a95-a6aa-3514d1ea3a10.png">
<img width="1416" alt="Screenshot 2021-11-15 at 10 15 10" src="https://user-images.githubusercontent.com/11256663/141754584-9984eb31-2143-4df4-9f6a-f46031e25378.png">
<img width="1642" alt="Screenshot 2021-11-15 at 10 15 25" src="https://user-images.githubusercontent.com/11256663/141754621-780c60e8-aeb2-48ef-b41d-95e4587d9ee5.png">

Here's the exported app - you can import this into your builder and see how I have done it.

[Input App-export-1636967795273.txt](https://github.com/Budibase/budibase/files/7537377/Input.App-export-1636967795273.txt)
Thanks!

**Reply by Romska** [reply]
Thank you for the detailed and quick answer! Using "global state" is a decision, but it seems a bit non flexible, but I will try it.

I hoped to hear, that there are other stored variables which can be accessed during navigateTo as parameteres of the function.
For example: navigateTo("main", {"input1":"hello", "input2":"{{name}}"}) 
And on main page you access them by something like {{ app.url.params['input1'] }}. 

Maybe it seems the same, but in this case it is transfered with reponse and you dont need to clear memory and free the variables.

**Reply by mjashanks** [reply]
> For example: navigateTo("main", {"input1":"hello", "input2":"{{name}}"})
And on main page you access them by something like {{ app.url.params['input1'] }}.

As @ggs331 suggested, you can also pass parameters via the URL.

First select your screen from the left-hand panel. Then you can change the URL to accept named parameters - 

![image](https://user-images.githubusercontent.com/3524181/141770936-cc51a5fc-133c-4266-824b-4d3497e67ca3.png)

Now when I edit this screen, I have access to the parameters for binding:

![image](https://user-images.githubusercontent.com/3524181/141771094-258af5ce-f40a-4e59-8143-63c06852d25f.png)

Then, in the `NavigateTo` action, we can pass the values, for example:
`/vehicles/hello/{{ name }}` - will set 
- `param1 = hello` 
- `param2 = {{ Form.Fields.Text Input }}` (binding from a form on the parent screen)

**Reply by Romska** [reply]
Thank you very much! I think this case is quite common and should be added to documentation one day.

The only think is that I dont want to transfer parameters in URL and in GET request with it, because values can be long and I dont want to display them for user.

May be it would be useful to add on screen panel "parameters" option.
